### ISSUE TITLE
Current metric name structure is hard to parse

### ISSUE THREAD

**Issue by cannium** [python-diamond/Diamond#188]
Current metric name structure is hard to parse for other applications like influxdb(see more explanations [here](https://github.com/influxdb/influxdb/issues/2102#issuecomment-101566403))
So I propose the metric name to be in a more uniformed structure, as `<prefix><hostname>[suffix]<metric name><measure>[device]`, for example, `servers.localhost.cpu.irq.cpu0`, `servers.desolation.loadavg.01` and `servers.desolation.iostat.reads.sda`(Originally they are `servers.localhost.cpu.cpu0.irq`, `servers.desolation.loadavg.01` and `servers.desolation.iostat.sda.reads`)
To do this, we need to patch some of the collectors. And I don't think these changes would break the use cases before.

**Comment by mattrobenolt** [comment]
Hmm, so this is tricky.

I definitely understand the need for changing this to better support influxdb, but at the expense of breaking everyone's existing metrics for graphite.

I wonder if there's a way we can come up with that collectors return back a string for the metric name, or optionally a future-proofed `namedtuple` that was split up into pieces. Then each outputter could control the exact output format. That way graphite can continue being the same way, and influxdb can be handled differently.

Make sense? Or is this overthinking?

**Comment by Investocat** [comment]
Wasn't this the goal with the metrics 2.0 (http://metrics20.org/) refactor? Is that still happening?

**Comment by josegonzalez** [comment]
So many refactors, so little time.

We should just cut a release of python-diamond now, start removing all the collectors, make a new release that supports external collectors, *then* start working on something that can resolve this. :(

**Comment by cannium** [comment]
@mattrobenolt I think all we need is a standard for metric names, currently collectors just output arbitrary names. For implementation, a `namedtuple` is nice, and it should have some kind of schema.

@josegonzalez I understand that you have your priorities. But is there a rough time schedule we could expect? I'll come to help if you need.

**Comment by josegonzalez** [comment]
@cannium see #181. We have no time-table, its up to everyone to help get this project back on track :)

**Comment by DStape** [comment]
http://metrics20.org (not sure how old/active/relevant this still is, but worth a read)
